File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if s is None or len(s) == 0:
            return True
        
        left = 0
        right = len(s) - 1

        while left <= right:
            charLeft = s[left]
            charRight = s[right]

            if not charLeft.isalpha() and not charLeft.isnumeric():
                left += 1
            elif not charRight.isalpha() and not charRight.isnumeric():
                right -= 1
            elif charLeft.lower() != charRight.lower():
                return False
            else:
                left += 1
                right -= 1
        
        return True

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {

        if (s == null || s.length() == 0) {
            return true;
        }

        int left = 0;
        int right = s.length() - 1;
        System.out.println(right);
        while (left < right) {
            char charLeft = s.charAt(left);
            char charRight = s.charAt(right);
            System.out.println(charLeft);
            System.out.println(charRight);

            if (!Character.isAlphabetic(charLeft) && !Character.isDigit(charLeft)) {
                left++;
            } else if (!Character.isAlphabetic(charRight) && !Character.isDigit(charRight)) {
                right--;
            } else if (Character.toLowerCase(charLeft) != Character.toLowerCase(charRight)) {

                return false;
            } else {
                left++;
                right--;
            }

        }

        return true;
    }
}